pkg/azuredx/datasource.go

```diff
@@ -124,12 +127,24 @@ func (plugin *GrafanaAzureDXDatasource) handleQuery(q backend.DataQuery, user *b
 		resp.Error = err
 	}
 
+	headers := map[string]string{}
+	msClientRequestIDHeader := fmt.Sprintf("KGC.%v;%v", qm.QuerySource, uuid.Must(uuid.NewRandom()).String())
+	if adx.settings.EnableUserTracking {
+		if user != nil {
+			msClientRequestIDHeader += fmt.Sprintf(";%v", user.Login)
+			headers["x-ms-user-id"] = user.Login
+		}
+	}
+	headers["x-ms-client-request-id"] = msClientRequestIDHeader
+
 	var tableRes *models.TableResponse
-	tableRes, kustoError, err = plugin.client.KustoRequest(plugin.settings, models.RequestPayload{
-		CSL:        qm.Query,
-		DB:         qm.Database,
-		Properties: models.NewConnectionProperties(plugin.settings, cs),
-	}, qm.QuerySource, user)
+	tableRes, _, kustoError, err = adx.client.KustoRequest(adx.settings.ClusterURL+"/v1/rest/query", models.RequestPayload{
```

Comment:
I see that we are concatenating urls with `+` everywhere but for future reference `ResolveReference` should be a more robust solution: https://golang.org/pkg/net/url/#URL.ResolveReference (it would avoid, for example, double slashes)

Reply:
yes good idea! we're gonna do lots of refactoring to the adx backend in the next few months. this is certainly one of those things!